src/app/modules/main/chat_section/controller.nim

```diff
@@ -162,18 +166,33 @@ method setActiveItemSubItem*(self: Controller, itemId: string, subItemId: string
 
   self.delegate.activeItemSubItemSet(self.activeItemId, self.activeSubItemId)
 
-method removeActiveFromThisChat*(self: Controller, itemId: string) =
-  if self.activeItemId != itemId:
-    return
+method deleteChannel*(self: Controller, itemId: string) =
+  let allChats = self.communityService.getChats(self.getMySectionId())
+  self.activeSubItemId = ""
+  if allChats.len == 0:
+    self.activeItemId = ""
+  else:
+    self.activeItemId = self.getMySectionId() & allChats[0].id
+
+  self.communityService.deleteCommunityChat(self.getMySectionId(), itemId)
 
+method deleteChat*(self: Controller, itemId: string) =
   let allChats = self.chatService.getAllChats()
-
   self.activeSubItemId = ""
   if allChats.len == 0:
     self.activeItemId = ""
   else:
     self.activeItemId = allChats[0].id
 
+  echo "Todo for chat "
+
+method removeChat*(self: Controller, itemId: string) =
```

Comment:
@Khushboo-dev-cpp since chat/channel may be deleted through the context menu even it is not the active one in that moment (so no need to deal with `self.activeItemId` and `self.activeSubItemId`) you can freely remove `deleteChannel` and `deleteChat` from this controller and use passed `itemId` and handle everything in a few lines within this method, something like this:
```
method removeChat*(self: Controller, itemId: string) =
  if self.isCommunitySection:
    self.communityService.deleteCommunityChat(self.getMySectionId(), itemId)
  else:
    self.chatService.deleteChatById(itemId) #this is not implemented yet I think
```